packages/astro/src/@types/astro.ts

```diff
@@ -2480,6 +2485,10 @@ export interface SSRResult {
  */
 export interface SSRMetadata {
 	hasHydrationScript: boolean;
+	/**
+	 * Keep track of which renderers already injected their specific hydration script
```

Comment:
Noted. I am editing the comment. Also, I'm going to change it to a `Set<string>` as this matches the pattern of `hasDirectives` and `propagators` which might make it easier to understand.

The naming is tricky. The name `hydrationScripts` could almost work, however it may be too confusing next to the unrelated `hasHydrationScript`. I think that's why @johannesspohr, who helped write this part of the PR, originally added `RendererSpecific`, to make it more clearly different from `hasHydrationScript`.

Thinking of it as a `Set`, I'm planning to revise it slightly to `rendererSpecificHydrationScripts`. This lets we rewrite the code in `packages/astro/src/runtime/server/render/common.ts` in a slightly easier to read way:

```ts
const { rendererSpecificHydrationScripts } = result._metadata;
const { rendererName } = instruction;

if (!rendererSpecificHydrationScripts.has(rendererName)) {
  rendererSpecificHydrationScripts.add(rendererName);
  return instruction.render();
}
return '';
```